lib/mayaUsd/ufe/UsdUndoMaterialCommands.cpp

```diff
@@ -112,6 +112,74 @@ bool _BindMaterialCompatiblePrim(const UsdPrim& usdPrim)
     return false;
 }
 
+//! Returns true if \p item is a materials scope.
+bool isMaterialsScope(const Ufe::SceneItem::Ptr& item)
+{
+    if (!item) {
+        return false;
+    }
+
+    // Must be a scope.
+    if (item->nodeType() != "Scope") {
+        return false;
+    }
+
+    // With the magic name.
+    if (item->nodeName() == UsdMayaJobExportArgs::GetDefaultMaterialsScopeName()) {
+        return true;
+    }
+
+    // Or with only materials inside
+    auto scopeHierarchy = Ufe::Hierarchy::hierarchy(item);
+    if (scopeHierarchy) {
+        for (auto&& child : scopeHierarchy->children()) {
+            if (child->nodeType() != "Material") {
+                // At least one non material
+                return false;
+            }
+        }
+    }
+
+    return true;
+}
+
+//! Searches the children of \p parentPath for a materials scope. Returns a null pointer if no
+//! materials scope is found.
+Ufe::SceneItem::Ptr getMaterialsScope(const Ufe::Path& parentPath)
```

Comment:
Factored out from `UsdUndoAssignNewMaterialCommand`. It is now being called by the new `UsdUndoCreateMaterialsScopeCommand`. This function is currently only called once but I still made it a separate function, as it could be useful for others at some point. This way it is easy to move it to a shared location if necessary.